scripts/build_energy_totals.py

```diff
@@ -737,6 +840,140 @@ def build_transport_data(countries, population, idees):
+        "electricity services water",
+        "electricity services cooking",
+        "electricity services",
+    ]
+
+    agri_total = [
+        "total agriculture heat",
+        "total agriculture machinery",
+        "total agriculture",
+    ]
+    agri_ele = [
+        "total agriculture electricity",
+    ]
+
+    road_total = [
+        "total road",
+        "total passenger cars",
+        "total other road passenger",
+        "total light duty road freight",
+    ]
+    road_ele = [
+        "electricity road",
+        "electricity passenger cars",
+        "electricity other road passenger",
+        "electricity light duty road freight",
+    ]
+
+    rail_total = [
+        "total rail",
+        "total rail passenger",
+        "total rail freight",
+    ]
+    rail_ele = [
+        "electricity rail",
+        "electricity rail passenger",
+        "electricity rail freight",
+    ]
+
+    avia_inter = [
+        "total aviation passenger",
+        "total aviation freight",
+        "total international aviation passenger",
+        "total international aviation freight",
+        "total international aviation",
+    ]
+    avia_domestic = [
+        "total domestic aviation passenger",
+        "total domestic aviation freight",
+        "total domestic aviation",
+    ]
+    navigation = [
+        "total domestic navigation",
+    ]
+
+    idees_countries = idees_countries.repalce({"GB": "UK", "GR": "EL"})
+
+    for country in idees_countries:
+        res = ratio.loc[(country, slice(None), "Residential")]
+        energy.loc[country, residential_total] *= res[["total"]].iloc[0, 0]
+        energy.loc[country, residential_ele] *= res[["ele"]].iloc[0, 0]
+
+        ser = ratio.loc[(country, slice(None), "Services")]
+        energy.loc[country, service_total] *= ser[["total"]].iloc[0, 0]
+        energy.loc[country, service_ele] *= ser[["ele"]].iloc[0, 0]
+
+        agri = ratio.loc[(country, slice(None), "Agriculture & forestry")]
+        energy.loc[country, agri_total] *= agri[["total"]].iloc[0, 0]
+        energy.loc[country, agri_ele] *= agri[["ele"]].iloc[0, 0]
+
+        road = ratio.loc[(country, slice(None), "Road")]
+        energy.loc[country, road_total] *= road[["total"]].iloc[0, 0]
+        energy.loc[country, road_ele] *= road[["ele"]].iloc[0, 0]
+
+        rail = ratio.loc[(country, slice(None), "Rail")]
+        energy.loc[country, rail_total] *= rail[["total"]].iloc[0, 0]
+        energy.loc[country, rail_ele] *= rail[["ele"]].iloc[0, 0]
+
+        avi_d = ratio.loc[(country, slice(None), "Domestic aviation")]
+        avi_i = ratio.loc[(country, "International aviation", slice(None))]
+        energy.loc[country, avia_inter] *= avi_i[["total"]].iloc[0, 0]
+        energy.loc[country, avia_domestic] *= avi_d[["total"]].iloc[0, 0]
+
+        nav = ratio.loc[(country, slice(None), "Domestic Navigation")]
+        energy.loc[country, navigation] *= nav[["total"]].iloc[0, 0]
+
+    return energy
```

Comment:
If you would want to have this more compact, you could also use such a structure:

```python
def rescale_energy(idees_countries, energy, eurostat):
    # ... eurostat calculations ...  

    country_mapping = {"GB": "UK", "GR": "EL"}
    idees_countries.replace(country_mapping, inplace=True)

    mappings = {  # Define your mappings here
        "residential": {
           "total": ["total residential space", ...],
           "ele": ["electricity residential space", ...]
        }, 
        "services": { ... },
        # ...
    }

    for country in idees_countries:
        for sector, mapping in mappings.items():
            sector_ratio = ratio.loc[(country, slice(None), sector)]

            # Vectorized scaling
            energy.loc[country, mapping['total']] *= sector_ratio['total'].iloc[0, 0]
            energy.loc[country, mapping['ele']] *= sector_ratio['ele'].iloc[0, 0]

    return energy
```

Reply:
Thanks for the tip :)